flake: agent/agentexec: TestExec/Linux

Seen running `make test` locally in dogfood:

```
=== FAIL: agent/agentexec TestExec/Linux/Enabled (0.00s)
    exec_test.go:64: 
                Error Trace:    /home/coder/src/coder/coder/agent/agentexec/exec_test.go:64
                Error:          Not equal: 
                                expected: []string{"/tmp/go-build791694279/b1047/agentexec.test", "agent-exec", "--", "sh", "-c", "sleep"}
                                actual  : []string{"/tmp/go-build791694279/b1047/agentexec.test", "agent-exec", "--coder-oom=10", "--coder-nice=1", "--", "sh", "-c", "sleep"}
                            
                                Diff:
                                --- Expected
                                +++ Actual
                                @@ -1,4 +1,6 @@
                                -([]string) (len=6) {
                                +([]string) (len=8) {
                                  (string) (len=43) "/tmp/go-build791694279/b1047/agentexec.test",
                                  (string) (len=10) "agent-exec",
                                + (string) (len=14) "--coder-oom=10",
                                + (string) (len=14) "--coder-nice=1",
                                  (string) (len=2) "--",
                Test:           TestExec/Linux/Enabled
        --- FAIL: TestExec/Linux/Enabled (0.00s)

=== FAIL: agent/agentexec TestExec/Linux/Nice (0.00s)
    exec_test.go:81: 
                Error Trace:    /home/coder/src/coder/coder/agent/agentexec/exec_test.go:81
                Error:          Not equal: 
                                expected: []string{"/tmp/go-build791694279/b1047/agentexec.test", "agent-exec", "--coder-nice=10", "--", "sh", "-c", "sleep"}
                                actual  : []string{"/tmp/go-build791694279/b1047/agentexec.test", "agent-exec", "--coder-oom=10", "--coder-nice=10", "--", "sh", "-c", "sleep"}
                            
                                Diff:
                                --- Expected
                                +++ Actual
                                @@ -1,4 +1,5 @@
                                -([]string) (len=7) {
                                +([]string) (len=8) {
                                  (string) (len=43) "/tmp/go-build791694279/b1047/agentexec.test",
                                  (string) (len=10) "agent-exec",
                                + (string) (len=14) "--coder-oom=10",
                                  (string) (len=15) "--coder-nice=10",
                Test:           TestExec/Linux/Nice
        --- FAIL: TestExec/Linux/Nice (0.00s)

=== FAIL: agent/agentexec TestExec/Linux/OOM (0.00s)
    exec_test.go:98: 
                Error Trace:    /home/coder/src/coder/coder/agent/agentexec/exec_test.go:98
                Error:          Not equal: 
                                expected: []string{"/tmp/go-build791694279/b1047/agentexec.test", "agent-exec", "--coder-oom=123", "--", "sh", "-c", "sleep"}
                                actual  : []string{"/tmp/go-build791694279/b1047/agentexec.test", "agent-exec", "--coder-oom=123", "--coder-nice=1", "--", "sh", "-c", "sleep"}
                            
                                Diff:
                                --- Expected
                                +++ Actual
                                @@ -1,2 +1,2 @@
                                -([]string) (len=7) {
                                +([]string) (len=8) {
                                  (string) (len=43) "/tmp/go-build791694279/b1047/agentexec.test",
                                @@ -4,2 +4,3 @@
                                  (string) (len=15) "--coder-oom=123",
                                + (string) (len=14) "--coder-nice=1",
                                  (string) (len=2) "--",
                Test:           TestExec/Linux/OOM
        --- FAIL: TestExec/Linux/OOM (0.00s)
```

--- a/agent/agentexec/cli_linux_test.go
+++ b/agent/agentexec/cli_linux_test.go
@@ -22,12 +22,9 @@ import (
 	"github.com/coder/coder/v2/testutil"
 )
 
+//nolint:paralleltest // This test is sensitive to environment variables
 func TestCLI(t *testing.T) {
-	t.Parallel()
-
 	t.Run("OK", func(t *testing.T) {
-		t.Parallel()
-
 		ctx := testutil.Context(t, testutil.WaitMedium)
 		cmd, path := cmd(ctx, t, 123, 12)
 		err := cmd.Start()
@@ -40,8 +37,6 @@ func TestCLI(t *testing.T) {
 	})
 
 	t.Run("Defaults", func(t *testing.T) {
-		t.Parallel()
-
 		ctx := testutil.Context(t, testutil.WaitMedium)
 		cmd, path := cmd(ctx, t, 0, 0)
 		err := cmd.Start()

--- a/agent/agentexec/exec_test.go
+++ b/agent/agentexec/exec_test.go
@@ -36,7 +36,7 @@ func TestExec(t *testing.T) {
 		//nolint:paralleltest // we need to test environment variables
 		t.Run("Disabled", func(t *testing.T) {
 			if runtime.GOOS != "linux" {
-				t.Skip("skipping on linux")
+				t.Skip("skipping on non-linux")
 			}
 
 			cmd, err := agentexec.CommandContext(context.Background(), "sh", "-c", "sleep")
@@ -52,7 +52,7 @@ func TestExec(t *testing.T) {
 			t.Setenv(agentexec.EnvProcPrioMgmt, "hello")
 
 			if runtime.GOOS != "linux" {
-				t.Skip("skipping on linux")
+				t.Skip("skipping on non-linux")
 			}
 
 			executable, err := os.Executable()
@@ -69,7 +69,7 @@ func TestExec(t *testing.T) {
 			t.Setenv(agentexec.EnvProcNiceScore, "10")
 
 			if runtime.GOOS != "linux" {
-				t.Skip("skipping on linux")
+				t.Skip("skipping on non-linux")
 			}
 
 			executable, err := os.Executable()
@@ -86,7 +86,7 @@ func TestExec(t *testing.T) {
 			t.Setenv(agentexec.EnvProcOOMScore, "123")
 
 			if runtime.GOOS != "linux" {
-				t.Skip("skipping on linux")
+				t.Skip("skipping on non-linux")
 			}
 
 			executable, err := os.Executable()
@@ -104,7 +104,7 @@ func TestExec(t *testing.T) {
 			t.Setenv(agentexec.EnvProcNiceScore, "14")
 
 			if runtime.GOOS != "linux" {
-				t.Skip("skipping on linux")
+				t.Skip("skipping on non-linux")
 			}
 
 			executable, err := os.Executable()

